swfiles/sw_spec2MDHisto.m

```diff
@@ -0,0 +1,267 @@
+function sw_spec2MDHisto(spectra,proj, dproj, filename)
+% saves spectrum to MDHisto
+% 
+% ### Syntax
+% 
+% sw_spec2MDHisto(spectra,proj,dproj,filename)`
+% 
+% ### Description
+% 
+% `sw_spec2MDHisto(spectra,proj,dproj,filename)` saves a 
+% spectrum that is calculated by sw_egrid
+%  
+% ### Input Arguments
+% spectra: a structure calculated by sw_egrid 
+% 
+% proj: a 3x3 matrix defining an orthogonal coordinate system 
+%       where each column is a vector defining the orientation 
+%       of the view. One of the vectors must be along a q axis 
+%       defined by the direction of the calculation.  
+% 
+% dproj: is a 3 vector that is the bin size in each of the 
+%        directions defined in proj. For the direction of the 
+%        calculation, this should be the step size.  
+%       
+% filename: is the name of the nexus file.  It will overwrite the existing
+%           file if one already exists
+%
+% Example:
+% q0 = [0 0 0];
+% qdir = [1 0 0];
+% nsteps = 100;
+% spec = sw_egrid(spinwave(sw_model('triAF', 1), {q0 q0+qdir nsteps}))
+% proj = [qdir(:) [0 1 0]' [0 0 1]'];
+% dproj = [(qdir(1)-q0(1)/steps, 1e-6, 1e-6];
+% sw_spec2MDHisto(spec, proj, dproj, 'testmdh.nxs');
+% Note that: 
+% (1) In the call to `spinwave`, only one q-direction may be specified
+%    e.g. the HKL specifier must be of the form {q0 q0+qdir nsteps} 
+% (2) one column in the `proj` matrix must be the q-direction used in 
+%    `spinwave` (e.g. `qdir`).
+
+
+if nargin==0
```

Comment:
We typically add a unit test for this like so
https://github.com/SpinW/spinw/blob/49880bc38bc853424a8cee431ce36fb5cdde4b55/%2Bsw_tests/%2Bunit_tests/unittest_spinw_addatom.m#L33-L40